utilities/imgls.pl

```diff
@@ -0,0 +1,603 @@
+#!/usr/bin/perl
+
+# imgls.pl: ls(1) listing supplemented with image thumbnails and dimensions
+#
+#    Usage: imgls [--width #] [--height #] [--[no]preserve_ratio]
+#                 [--[no]dimensions] [--[no]unknown] [ls options] [file ...]
+#
+# Writing an image to an iTerm window is simple. See the official documentation
+# at https://iterm2.com/documentation-images.html and the write_image() function below.
+#
+# Many of ls' options are supported, but not all.  The program does not support ls'
+# default -C columnar output mode - output is always one entry per line.  Writing
+# images across the page in columns appears to be problematic.
+#
+# In addition, options are available to specify setting image properties (width, height,
+# preserve aspect ratio), include inline image dimensions, and disable output of
+# generic icons for unsupported image types. Finally, a table-based image dimensions
+# lookup mechanism is employed to obtain image dimensions.  It can call on Perl
+# modules such as Image::Size, or call on external programs such as sips, mdls or php.
+# It is easy to add additional entries to the table.  You can use the --method option
+# to select any of the currently supported methods ('sips', 'mdls', 'php', and
+# 'image::size').  These are tried, in that order; the first that appears to work
+# is used for all.
+
+use v5.14;
+use strict;
+use utf8;
+use warnings;
+
+use File::stat;
+use IO::Select;
+use IPC::Open3;
+use File::Which;
```

Comment:
This looks good. The only troublesome dependency left is `File::Which`. CentOS 7 is the most hostile environment we run in and a default `yum install perl` doesn't give you this. It's not a dealbreaker, but if it's easy to remove this dependency it will save me some pain down the line.

Reply:
Let me know what you think of this: https://gist.github.com/gnachman/0c18fb284da398f9f83d2b18da5692a1

Reply:
The code is updated with a suitable which() replacement.